src/App.jsx

```diff
@@ -14,33 +18,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const getPreparedGoods = (goods, { sortField, isReversed }) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleClear = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const isChanged = sortField || isReversed;
+
+  const goods = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
+  const buttons = [
+    {
+      name: 'Sort alphabetically',
+      className: `is-info ${sortField === SORT_BY_ALPHABET ? '' : 'is-light'}`,
+      handleClick() {
+        setSortField(SORT_BY_ALPHABET);
+      },
+    },
+    {
+      name: 'Sort by length',
+      className: `is-success ${sortField === SORT_BY_LENGTH ? '' : 'is-light'}`,
+      handleClick() {
+        setSortField(SORT_BY_LENGTH);
+      },
+    },
+    {
+      name: 'Reverse',
+      className: `is-warning ${isReversed ? '' : 'is-light'}`,
+      handleClick() {
+        setIsReversed(!isReversed);
```

Comment:
```suggestion
        setIsReversed(prevState => !prevState);
```
Use functional `setState` when you need to update state depending on the previous one